---
title: "CeDExplorer Tutorial"
author: "Yuxi Zhang"
date: "2025-01-01"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{CeDExplorer Tutorial}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

# CeDExplorer Tutorial

## Introduction

CeDExplorer provides comprehensive tools for celiac disease research, integrating GWAS analysis, gene expression profiling, and network visualization.

## Installation and Loading

```{r, message=FALSE}
library(CeDExplorer)
```

## GWAS Analysis

### Get GWAS Data
```{r}
library(CeDExplorer)
gwas_data <- getCeliacGwas(fetch = TRUE)
```
Fetches all celiac-diseaseâ€“related SNPs from GWAS Catalog

Includes genomic position, p-values, risk alleles, and mapped genes


### Top GWAS Hits and GWAS Summary
```{r}
plotTopGwasHits(gwas_data, top_n = 15, 
                   title = "Top Celiac Disease GWAS Associations")
plotGwasSummary(gwas_data)
```

## Gene Expression Analysis

```{r}
expressionBoxplot("HLA-DQA1", violin = TRUE)
```

## Network Analysis

```{r}
genes <- c("HLA-DQA1", "HLA-DQB1", "IL2", "STAT1")
plotGeneNetwork(genes)
```

## HLA Associations

```{r}
hlaInteractionPlot("HLA-DQA1")
```

## Composite Dashboard

```{r}
geneEvidenceView("HLA-DQA1")
```
```{r, message=FALSE}
library(CeDExplorer)
